src/generate_form.jsx

```diff
@@ -0,0 +1,306 @@
+const config = {
+    title: "HCC Calculator",
+    info: "A post-transplant hepatocellular carcinoma (HCC) recurrence calculator using a machine-learning algorithm mapped on pre-operative patient and tumor characteristics",
+    roundoff: 2
+}
+
+const variables = [{
+    label: "Pre-transplant neutrophil count",
+    name: "nCnt",
+    type: "Float",
+    display: "Input",
+    weight: 0.1
+},{
+    label: "Number of bridging treatments",
+    name: "nBT",
+    type: "Integer",
+    display: "Dropdown",
+    upper: 8,
+    weight: 0.2
+},{
+    label: "Number of PEI treatments",
+    name: "nPEI",
+    type: "Integer",
+    display: "Dropdown",
+    upper: 12,
+    weight: 0.3
+},{
+    label: "Size of largest lesion at time of listing",
+    name: "lSize",
+    type: "Float",
+    display: "Input",
+    weight: 0.4
+},{
+    label: "Total tumor diameter at the time of listing",
+    name: "tDiameter",
+    type: "Float",
+    display: "Input",
+    unit: "cm",
+    weight: 0.5
+}];
+
+import { 
+    Button,
+    Dialog,
+    DialogContent,
+    Grid,
+    InputAdornment,
+    makeStyles,
+    MenuItem,
+    Select,
+    TextField,
+    Tooltip,
+    Typography
+} from '@material-ui/core';
+import InfoIcon from '@material-ui/icons/Info';
+import ReactDOM from "react-dom";
+import React from "react";
+
+const useSelectStyles = makeStyles(theme => ({
+    root: {
+        display: 'flex',
+        flexWrap: 'wrap',
+    },
+    select: {
+        minWidth: 320
+    },
+}));
+
+const useTextFieldStyles = makeStyles(theme => ({
+    textfield: {
+        minWidth: 320
+    }
+}));
+
+const useButtonStyles = makeStyles(theme => ({
+    button: {
+        margin: theme.spacing(1),
+    },
+    input: {
+        display: 'none'
+    },
+}));
+
+const useTooltipStyles = makeStyles(theme => ({
+    tooltip: {
+        color: "white",
+        backgroundColor: "black"
+    }
+}));
+
+function EnterData(props) {
+    const [val, setVal] = React.useState('');
+    function handleChange(event) {
+        setVal(event.target.value);
+        props.updateParent(props.metadata.name, event.target.value);
+    }
+
+    if (props.metadata.display == "Input") {
+        const textfield_classes = useTextFieldStyles();
+        return (
+            <TextField
+                className={textfield_classes.textfield}
+                id={props.metadata.name}
+                type="number"
+                onChange={handleChange}
+                error={props.err}
+                margin="dense"
+                width="auto"
+                InputProps={{
+                    endAdornment: <InputAdornment position="end">{(props.metadata.hasOwnProperty('unit') ? props.metadata.unit : "")}</InputAdornment>
```

Comment:
Shorter and with the same effect:
```suggestion
                    endAdornment: <InputAdornment position="end">{(props.metadata.unit || "")}</InputAdornment>
```